expapi/v2/metrics_handlers.go

```diff
@@ -0,0 +1,190 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	"errors"
+	"net/http"
+	"time"
+
+	restful "github.com/emicklei/go-restful"
+	"github.com/golang/glog"
+
+	"k8s.io/heapster/expapi/v2/types"
+	model_api "k8s.io/heapster/model"
+	"k8s.io/heapster/store/statstore"
+	"k8s.io/kubernetes/pkg/api/resource"
+)
+
+// errModelNotActivated is the error that is returned when manager.cluster
+// has not beed initialized.
+var errModelNotActivated = errors.New("the model is not activated")
+
+// RegisterMetrics registers the Metrics API endpoints.
+// All endpoints that end with a {metric-name} also receive a start time query parameter.
+// The start and end times should be specified as a string, formatted according to RFC 3339.
+func (a *Api) RegisterMetrics(container *restful.Container) {
+	ws := new(restful.WebService)
+	ws.
+		Path("/experimental/v2").
+		Doc("Root endpoint of the stats model").
+		Consumes("*/*").
+		Produces(restful.MIME_JSON)
+
+	ws.Route(ws.GET("/nodeMetrics/available/").
+		To(a.availableMetrics).
+		Filter(compressionFilter).
+		Doc("Get a list of all available metrics for a Node entity").
+		Writes([]string{}).
+		Operation("availableMetrics"))
+
+	ws.Route(ws.GET("/nodeMetrics/derived/").
+		To(a.derivedNodeMetricsList).
+		Filter(compressionFilter).
+		Doc("Get a list of all available metrics for all nodes").
+		Writes(types.DerivedNodeMetricsList{}).
+		Operation("derivedNodeMetricsList"))
+
+	ws.Route(ws.GET("/nodeMetrics/derived/{node-name}").
+		To(a.derivedNodeMetrics).
+		Filter(compressionFilter).
+		Doc("Get a list of all raw metrics for a Node entity").
+		Writes(types.DerivedNodeMetrics{}).
+		Operation("derivedNodeMetrics").
+		Param(ws.PathParameter("node-name", "The name of the node to look up").DataType("string")))
+
+	container.Add(ws)
+}
+
+// availableMetrics returns a list of available metric names.
+// These metric names can be used to extract metrics from the various model entities.
+func (a *Api) availableMetrics(request *restful.Request, response *restful.Response) {
+	model := a.manager.GetModel()
+	if model == nil {
+		response.WriteError(400, errModelNotActivated)
+		return
+	}
+	result := model.GetAvailableMetrics()
+	response.WriteEntity(result)
+}
+
+func getNodeMetrics(model model_api.Model, name string) (*types.DerivedNodeMetrics, error) {
+	res, _, err := model.GetNodeStats(model_api.NodeRequest{
+		NodeName: name,
+	})
+	if err != nil {
+		return nil, err
+	}
+	const minute, hour, day = "1m", "1h", "1d"
```

Comment:
Sure.

BTW, this is a WIP, so don't review yet as this is still being written.